core/storage/feedback/gae_models.py

```diff
@@ -420,6 +421,400 @@ def get_multi(cls, user_id, thread_ids):
         return super(FeedbackThreadUserModel, cls).get_multi(instance_ids)
 
 
+class GeneralFeedbackThreadModel(base_models.BaseModel):
+    """Threads for each entity.
+
+    The id of instances of this class has the form
+        [ENTITY_TYPE].[ENTITY_ID].[GENERATED_STRING]
+    """
+    # The type of entity the thread is linked to.
+    entity_type = ndb.StringProperty(required=False, indexed=True)
+    # The ID of the entity the thread is linked to.
+    entity_id = ndb.StringProperty(required=False, indexed=True)
+    # ID of the user who started the thread. This may be None if the feedback
+    # was given anonymously by a learner.
+    original_author_id = ndb.StringProperty(indexed=True)
+    # Latest status of the thread.
+    status = ndb.StringProperty(
+        default=STATUS_CHOICES_OPEN,
+        choices=STATUS_CHOICES,
+        required=True,
+        indexed=True,
+    )
+    # Latest subject of the thread.
+    subject = ndb.StringProperty(indexed=False)
+    # Summary text of the thread.
+    summary = ndb.TextProperty(indexed=False)
+    # Specifies whether this thread has a related learner suggestion.
+    has_suggestion = ndb.BooleanProperty(indexed=True, default=False)
+    # The number of messages in the thread.
+    message_count = ndb.IntegerProperty(indexed=True)
+    # When this thread was last updated. This overrides the field in
+    # BaseModel. We are overriding it because we do not want the last_updated
+    # field to be updated everytime the feedback thread is changed. For example,
+    # on running the job for calculating the number of messages in a thread
+    # and updating the message_count field we do not wish the last_updated field
+    # to be updated.
```

Comment:
But after migration, you do. So add a TODO and file an issue to get rid of this after the migration (which I'm assuming we'll only do once and then delete the old models and migration jobs?).

Reply:
Actually here we do need it. A lot of manual manipulation of the last updated field is done.. For example, when a message is added, the thread's last updated field is updated, when message count is updated by some job, last_updated is not updated (though the thread is updated), and some more situations. For the message model, I have added a TODO.

Reply:
OK. Thanks for the explanation.